model2vec/train/classifier.py

```diff
@@ -66,14 +69,32 @@ def construct_head(self) -> nn.Sequential:
 
         return nn.Sequential(*modules)
 
-    def predict(self, X: list[str], show_progress_bar: bool = False, batch_size: int = 1024) -> np.ndarray:
-        """Predict a class for a set of texts."""
-        pred: list[str] = []
+    def predict(
+        self, X: list[str], show_progress_bar: bool = False, batch_size: int = 1024, threshold: float = 0.5
+    ) -> np.ndarray:
+        """
+        Predict labels for a set of texts.
+
+        In single-label mode, each prediction is a single class.
+        In multilabel mode, each prediction is a list of classes.
+
+        :param X: The texts to predict on.
+        :param show_progress_bar: Whether to show a progress bar.
+        :param batch_size: The batch size.
+        :param threshold: The threshold for multilabel classification.
+        :return: The predictions.
+        """
+        pred = []
         for batch in trange(0, len(X), batch_size, disable=not show_progress_bar):
             logits = self._predict_single_batch(X[batch : batch + batch_size])
-            pred.extend([self.classes[idx] for idx in logits.argmax(1)])
-
-        return np.asarray(pred)
+            if self.multilabel:
+                probs = torch.sigmoid(logits)
```

Comment:
I think this could be clearer by first turning everything into a booltensor

```
above_threshold = probs > threshold
```

```
for sample in above_threshold:
    sample_labels = [self.classes[x] for x in torch.nonzero(sample).squeeze()]
```

If we turn classes into an array, we could also index immediately:

```
self.classes[torch.nonzero(sample).squeeze()]
```

Anyway, just a small optimization